backend/api/quivr_api/modules/sync/utils/dropboxutils.py

```diff
@@ -0,0 +1,308 @@
+from quivr_api.modules.upload.service.upload_file import check_file_exists
+
+logger = get_logger(__name__)
+
+APP_KEY = os.getenv("DROPBOX_APP_KEY")
+APP_SECRET = os.getenv("DROPBOW_CONSUMER_SECRET")
+
+
+class DropboxSyncUtils(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    storage: Storage
+    sync_files_repo: SyncFiles
+    sync_active_service: SyncService
+    sync_user_service: SyncUserService
+
+    async def _upload_files(
+        self,
+        token_data: dict,
+        files: list,
+        current_user: str,
+        brain_id: str,
+        sync_active_id: int,
+    ):
+        """
+        Download files from DropBox.
+
+        Args:
+            credentials (dict): The credentials for accessing Google Drive.
+            files (list): The list of file metadata to download.
+
+        Returns:
+            dict: A dictionary containing the status of the download or an error message.
+        """
+        dbx = dropbox.Dropbox(token_data["access_token"])
+        dbx.check_and_refresh_access_token()
+        token_data["access_token"] = dbx._oauth2_access_token
+
+        downloaded_files = []
+        bulk_id = uuid.uuid4()
+        for file in files:
+            logger.info("🔥🔥🔥🔥: %s", file)
+            try:
+                file_id = str(file["id"])
+                file_name = file["name"]
+                mime_type = file["mime_type"]
+                modified_time = file["last_modified"]
+
+                download_response = dbx.files_download(file_id)
+                file_data = BytesIO(download_response.content)
+
+                # Check if the file already exists in the storage
+                if check_file_exists(brain_id, file_name):
+                    logger.debug("🔥 File already exists in the storage: %s", file_name)
+
+                    self.storage.remove_file(brain_id + "/" + file_name)
+                    BrainsVectors().delete_file_from_brain(brain_id, file_name)
+
+                # Check if the file extension is compatible
+                if file_name.split(".")[-1] not in [
+                    "pdf",
+                    "txt",
+                    "md",
+                    "csv",
+                    "docx",
+                    "xlsx",
+                    "pptx",
+                    "doc",
+                ]:
+                    logger.info("File is not compatible: %s", file_name)
+                    continue
+
+                to_upload_file = UploadFile(
+                    file=file_data,
+                    filename=file_name,
+                )
+
+                # Check if the file already exists in the database
+                existing_files = self.sync_files_repo.get_sync_files(sync_active_id)
+                existing_file = next(
+                    (f for f in existing_files if f.path == file_name), None
+                )
+
+                supported = False
+                if (existing_file and existing_file.supported) or not existing_file:
```

Comment:
I see duplicate logic on the if/else statements. Is there a way to have a more compact branching logic here 🤔